plugin/trino-pinot/src/test/java/io/trino/plugin/pinot/AbstractPinotIntegrationSmokeTest.java

```diff
@@ -2203,4 +2226,24 @@ public void testVarbinary()
         assertThat(query("SELECT bytes_col FROM \"SELECT bytes_col, string_col FROM alltypes\" WHERE string_col != 'array_null'"))
                 .matches(expectedValues);
     }
+
+    @Test
+    public void testJson()
+    {
+        assertThat(query("SELECT string_col, json_col FROM " + JSON_TYPE_TABLE))
+                .matches("VALUES (VARCHAR 'string_0', JSON '{\"id\":0,\"name\":\"user_0\"}')," +
+                        "  (VARCHAR 'string_1', JSON '{\"id\":1,\"name\":\"user_1\"}')," +
+                        "  (VARCHAR 'string_2', JSON '{\"id\":2,\"name\":\"user_2\"}')");
+        assertThat(query("SELECT string_col, json_col" +
+                "  FROM \"SELECT string_col, json_col FROM " + JSON_TYPE_TABLE + "\""))
+                .matches("VALUES (VARCHAR 'string_0', JSON '{\"id\":0,\"name\":\"user_0\"}')," +
+                        "  (VARCHAR 'string_1', JSON '{\"id\":1,\"name\":\"user_1\"}')," +
+                        "  (VARCHAR 'string_2', JSON '{\"id\":2,\"name\":\"user_2\"}')");
+        assertThat(query("SELECT name FROM \"SELECT json_extract_scalar(json_col, '$.name', 'STRING', '0') AS name" +
+                "  FROM json_table WHERE json_extract_scalar(json_col, '$.id', 'INT', '0') = '1'\""))
+                .matches("VALUES (VARCHAR 'user_1')");
+        assertThat(query("SELECT JSON_EXTRACT(json_col, '$.name') FROM " + JSON_TYPE_TABLE +
+                "  WHERE JSON_EXTRACT(json_col, '$.id') = JSON '1'"))
+                .matches("VALUES (JSON '\"user_1\"')");
+    }
```

Comment:
It would be nice to add a test case for predicates. e.g.
```java
        assertThat(query("SELECT string_col FROM " + JSON_TYPE_TABLE + " WHERE json_col = JSON '{\"id\":0,\"name\":\"user_0\"}'"))
                .matches("VALUES 'string_0'");
```